pkg/webhook/validate.go

```diff
@@ -35,17 +37,60 @@ func validateSriovOperatorConfig(cr *sriovnetworkv1.SriovOperatorConfig, operati
 	glog.V(2).Infof("validateSriovOperatorConfig: %v", cr)
 	var warnings []string
 
-	if cr.GetName() == constants.DefaultConfigName {
-		if operation == v1.Delete {
-			return false, warnings, fmt.Errorf("default SriovOperatorConfig shouldn't be deleted")
+	if cr.GetName() != constants.DefaultConfigName {
+		return false, warnings, fmt.Errorf("only default SriovOperatorConfig is used")
+	}
+
+	if operation == v1.Delete {
+		return false, warnings, fmt.Errorf("default SriovOperatorConfig shouldn't be deleted")
+	}
+
+	if cr.Spec.DisableDrain {
+		warnings = append(warnings, "Node draining is disabled for applying SriovNetworkNodePolicy, it may result in workload interruption.")
+	}
+
+	err := validateSriovOperatorConfigDisableDrainWhileUpdating(cr)
+	if err != nil {
+		return false, warnings, err
+	}
+
+	return true, warnings, nil
+}
+
+// validateSriovOperatorConfigDisableDrainWhileUpdating checks if the user is setting `.Spec.DisableDrain` from false to true while
+// operator is updating one or more nodes. Disabling the drain at this stage would prevent the operator to uncordon a node at
+// the end of the update operation, keeping nodes un-schedulable until manual intervention.
+func validateSriovOperatorConfigDisableDrainWhileUpdating(cr *sriovnetworkv1.SriovOperatorConfig) error {
```

Comment:
nit: shorten it to: validateSriovOperatorConfigDisableDrain ?

Reply:
sure